llvm/lib/Transforms/Scalar/ConstraintElimination.cpp

```diff
@@ -1629,6 +1634,11 @@ static bool eliminateConstraints(Function &F, DominatorTree &DT, LoopInfo &LI,
 
     ICmpInst::Predicate Pred;
     if (!CB.isConditionFact()) {
+      if (Value *X; match(CB.Inst, m_Intrinsic<Intrinsic::abs>(m_Value(X)))) {
+        AddFact(CmpInst::ICMP_SGE, CB.Inst, X);
```

Comment:
I think we should also be able to add `CB.Inst >= 0` as a follow-up?

Reply:
Sounds good, could you add a TODO?

Reply:
Done